src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,97 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROWTH_INDEX = 1.5;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growArray();
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index: " + index);
+        }
+        growArray();
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
```

Comment:
I would suggest reusage of existing methods
```suggestion
        if (index == size) {
            add(value);
            return;
        }
        checkIndex(index);
        growArray();
        System.arraycopy(array, index, array, index + 1, size - index);
        array[index] = value;
        size++;
```